File: Algorithms/11_Container_With_Most_Water/Solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_area, l, r = 0, 0, len(height) - 1
        while l < r:
            max_area = max(max_area, (r - l) * min(height[l], height[r]))
            l, r = (l + 1, r) if height[l] < height[r] else (l, r - 1)
        return max_area

File: Algorithms/11_Container_With_Most_Water/Solution.kt
/**
 * @author occidere
 * @since 2019-05-30
 * Blog: https://blog.naver.com/occidere
 * Github: https://github.com/occidere
 */
class Solution {
    fun maxArea(height: IntArray): Int {
        var max = 0
        val len = height.size
        for (i in 0 until len - 1) {
            for (j in i + 1 until len) {
                val w = j - i
                val h = Math.min(height[i], height[j])
                max = Math.max(max, w * h)
            }
        }
        return max
    }
}
